mars/oscar/backends/ray/tests/test_ray_actor_context.py

```diff
@@ -0,0 +1,219 @@
+    def setUp(self) -> None:
+        ray.init()
+
+    def tearDown(self) -> None:
+        ray.shutdown()
+
+    async def testSimpleLocalActorPool(self):
+        actor_ref = await mo.create_actor(DummyActor, 100, address=RAY_TEST_ADDRESS)
+        self.assertEqual(await actor_ref.add(1), 101)
+        await actor_ref.add(1)
+
+        res = await actor_ref.get_value()
+        self.assertEqual(res, 102)
+
+        ref2 = await actor_ref.get_ref()
+        self.assertEqual(actor_ref.address, ref2.address)
+        self.assertEqual(actor_ref.uid, ref2.uid)
+
+        self.assertEqual(await mo.actor_ref(
+                uid=actor_ref.uid, address=actor_ref.address).add(2), 104)
+
+    async def testLocalPostCreatePreDestroy(self):
+        actor_ref = await mo.create_actor(EventActor, address=RAY_TEST_ADDRESS)
+        await actor_ref.destroy()
+
+    async def testLocalCreateActor(self):
+        actor_ref = await mo.create_actor(DummyActor, 1, address=RAY_TEST_ADDRESS)
+        # Fail on ray backend, the DummyActor.create will create an actor in actor process.
+        # mo.actor_ref can't get the actor in current process.
+        # create actor inside on_receive
+        # r = await actor_ref.create(DummyActor, 5, address=RAY_TEST_ADDRESS)
+        # ref = mo.actor_ref(r)
+        # self.assertEqual(await ref.add(10), 15)
+        # create actor inside on_receive and send message
+        r = await actor_ref.create_send(DummyActor, 5, method='add', method_args=(1,), address=RAY_TEST_ADDRESS)
+        self.assertEqual(r, 6)
+
+    async def testLocalCreateActorError(self):
+        ref1 = await mo.create_actor(DummyActor, 1, uid='dummy1', address=RAY_TEST_ADDRESS)
+        with self.assertRaises(mo.ActorAlreadyExist):
+            await mo.create_actor(DummyActor, 1, uid='dummy1', address=RAY_TEST_ADDRESS)
+        await mo.destroy_actor(ref1)
+
+        # It's hard to get the exception raised from Actor.__init__.
+        # with self.assertRaises(ValueError):
+        #     await mo.create_actor(DummyActor, -1, address=RAY_TEST_ADDRESS)
+        ref1 = await mo.create_actor(DummyActor, 1, address=RAY_TEST_ADDRESS)
+        with self.assertRaises(ValueError):
+            await ref1.create(DummyActor, -2)
+
+    async def testLocalSend(self):
+        ref1 = await mo.create_actor(DummyActor, 1)
+        ref2 = mo.actor_ref(await ref1.create(DummyActor, 2))
+        self.assertEqual(await ref1.send(ref2, 'add', 3), 5)
+
+    async def testLocalSendError(self):
+        ref1 = await mo.create_actor(DummyActor, 1)
+        with self.assertRaises(TypeError):
+            await ref1.add(1.0)
+        ref2 = await mo.create_actor(DummyActor, 2)
+        with self.assertRaises(TypeError):
+            await ref1.send(ref2, 'add', 1.0)
+        with self.assertRaises(mo.ActorNotExist):
+            await mo.actor_ref('fake_uid').add(1)
+
+    # Fail on ray backend, the DummyActor.create will create an actor in actor process.
```

Comment:
Configuring RayActorContext as the global context when starting an actor in Ray will enable creating actor in actors.

Reply:
The problem is that `DummyActor, 1` and `DummyActor, 2` are in different processes.
- The RayActorContext in current process has actor `DummyActor, 1`
- The RayActorContext in `DummyActor, 1` has actor `DummyActor, 2`.

Then we get `DummyActor, 2` in current process as `ref2`, any call on `ref2` will be failed. Maybe we can use ray named actor to fix this issue.